feat(sdn): add support for VNets datasource

maciej-lech opened the pull request:
### Contributor's Note
<!--- 
Please mark the following items with an [x] if they apply to your PR.
Leave the [ ] if they are not applicable, or if you have not completed the item.
--->
- [X] I have added / updated documentation in `/docs` for any user-facing features or additions.
- [X] I have added / updated acceptance tests in `/fwprovider/tests` for any new or updated resources / data sources.
- [X] I have ran `make example` to verify that the change works as expected.

<!---
You can find more information about coding conventions and local testing in the [CONTRIBUTING.md](https://github.com/bpg/terraform-provider-proxmox/blob/main/CONTRIBUTING.md) file.

If you are unsure how to run `make example`, see [Deploying the example resources](https://github.com/bpg/terraform-provider-proxmox?tab=readme-ov-file#deploying-the-example-resources) section in README.
--->

<!--
*IF* your code contains breaking changes make sure to add `!` to the end of commit type, e.g.:
```
    feat(vm)!: add support for new feature 
```
Also, uncomment the section just below, and add a description of the breaking change. 
--->

<!---
#### ⚠ BREAKING CHANGES

>>> Put your description here <<<
--->

### Proof of Work
1. `make testacc` some tests are failing, but I think these are unrelated (most probably related to my proxmox setup), although as for below error seems there is some problem in SDN subnet implementation.

```
=== NAME  TestAccDataSourceSDNSubnet
    datasource_test.go:24: Step 1/1 error: Error running apply: exit status 1
        
        Error: Value Conversion Error
        
          with data.proxmox_virtual_environment_sdn_subnet.datasource_subnet,
          on terraform_plugin_test.tf line 43, in data "proxmox_virtual_environment_sdn_subnet" "datasource_subnet":
          43: 				data "proxmox_virtual_environment_sdn_subnet" "datasource_subnet" {
        
        An unexpected error was encountered trying to convert into a Terraform value.
        This is always an error in the provider. Please report the following to the
        provider developer:
        
        Cannot use attr.Value types.IPAddrValue, only basetypes.StringValue is
        supported because basetypes.StringType is the type in the schema
--- FAIL: TestAccDataSourceSDNSubnet (1.48s)
```

Other than that all SDN-related succeeded.

[testacc.log](https://github.com/user-attachments/files/23260510/testacc.log)

2. `make example` succeeded

[example.log](https://github.com/user-attachments/files/23260682/example.log)

<!--- Please keep this note for the community --->
### Community Note

- Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
<!--- Thank you for keeping this note for the community --->

<!--- If your PR fully resolves and should automatically close the linked issue, use Closes. Otherwise, use Relates --->
Closes #2211 

<!--- Release note for [CHANGELOG](https://github.com/bpg/terraform-provider-proxmox/blob/main/CHANGELOG.md) will be created automatically using the PR's title, update it accordingly. --->

maciej-lech commented:
/gemini review

maciej-lech commented on fwprovider/cluster/sdn/vnet/datasource_vnets.go:
```diff
@@ -0,0 +1,169 @@
+						"isolate_ports": types.BoolType,
+						"tag":           types.Int64Type,
+						"vlan_aware":    types.BoolType,
+					},
```
Duplicate

bpg commented:
> Ok, #2300 should fix subnet datasource issue. I also added there examples for SDN datasources. Once that one is merged I can update this PR to also include this new datasource in the examples.

Thanks! I'll merge #2300 first

maciej-lech commented:
Ok, the code is now rebased with `main` and I added this new datasource to the examples.

[example.log](https://github.com/user-attachments/files/23290746/example.log)
